vcs/git.go

```diff
@@ -99,7 +156,7 @@ func (g *GitDriver) Clone(dir, url string) (string, error) {
 		return "", err
 	}
 
-	return g.HeadRev(dir)
+	return g.Pull(dir)
```

Comment:
Can you give me more details on what this additional `Pull` does here?

Reply:
`Clone` initially checks out the default HEAD branch. The `Pull` ensures that we check out out the target ref.

Reply:
got it, much appreciated.